Makefile

```diff
@@ -139,23 +149,56 @@ OBJS += \
 	pow_fpga_accel.o
 endif
 
+ifeq ("$(BUILD_REMOTE)", "1")
+OBJS += \
+	remote_common.o \
+        remote_interface.o
+
+WORKER_OBJS := $(addprefix $(OUT)/remote/,$(filter-out remote_interface.o, $(OBJS)))
+OBJS := $(addprefix $(OUT)/, $(OBJS))
+LIBDCURL_OBJS := $(OBJS)
+OBJS += $(WORKER_OBJS)
+
+WORKER_CFLAGS := $(filter-out -DENABLE_REMOTE, $(CFLAGS))
+else 
 OBJS := $(addprefix $(OUT)/, $(OBJS))
+endif
 
 $(OUT)/test-%.o: tests/test-%.c $(LIBTUV_PATH)/include
 	$(VECHO) "  CC\t$@\n"
 	$(Q)$(CC) -o $@ $(CFLAGS) -I $(SRC) $(LIBTUV_INCLUDE) -c -MMD -MF $@.d $<
 
-$(OUT)/%.o: $(SRC)/%.c $(LIBTUV_PATH)/include
+$(OUT)/%.o: $(SRC)/%.c $(LIBTUV_PATH)/include $(LIBRABBITMQ_PATH)/build/include
 	$(VECHO) "  CC\t$@\n"
-	$(Q)$(CC) -o $@ $(CFLAGS) $(LIBTUV_INCLUDE) -c -MMD -MF $@.d $<
+	$(Q)$(CC) -o $@ $(CFLAGS) $(LIBTUV_INCLUDE) $(LIBRABBITMQ_INCLUDE) -c -MMD -MF $@.d $<
 
-$(OUT)/test-%: $(OUT)/test-%.o $(OBJS) $(LIBTUV_LIBRARY)
+ifeq ("$(BUILD_REMOTE)", "1")
```

Comment:
The conditional statement can be removed if the variable `LIBDCURL_OBJS` can be replaced with `OBJS`.

Reply:
`OBJS` is for building total objects including `LIBDCURL_OBJS` and `WORKER_OBJS`. `LIBDCURL_OBJS` is for linking them for `libdcurl.so`. `WORKER_OBJS` is for linking them for remote-worker.

Reply:
Okey, I try.